src/Api/Controller/AbstractSerializeController.php

```diff
@@ -222,26 +222,27 @@ protected function loadRelations(Collection $models, array $relations, ServerReq
             $relations = array_unique($relations);
         }
 
-        $callables = [];
+        $callableRelations = [];
+        $nonCallableRelations = [];
 
         foreach ($relations as $k => $relation) {
             if (isset($addedRelationCallables[$relation])) {
                 $load = $addedRelationCallables[$relation];
 
-                $callables[$relation] = function ($query) use ($load, $request, $relations) {
+                $callableRelations[$relation] = function ($query) use ($load, $request, $relations) {
                     $load($query, $request, $relations);
                 };
-
-                unset($relations[$k]);
+            } else {
+                $nonCallableRelations[$k] = $relation;
```

Comment:
What is $k here? Maybe we could just append directly without setting it to a particular key value? If so, that would simplify the for loop too.